ui/src/reusable_ui/components/button/Button.tsx

```diff
@@ -0,0 +1,96 @@
+// Libraries
+import React, {Component} from 'react'
+import classnames from 'classnames'
+
+// Types
+import {
+  ComponentColor,
+  ComponentSize,
+  ButtonShape,
+  IconFont,
+} from 'src/reusable_ui/types'
+import {RemoteDataState} from 'src/types/flux'
+
+// Styles
+import './Button.scss'
+
+interface Props {
+  text: string
+  onClick?: () => void
+  color?: ComponentColor
+  size?: ComponentSize
+  shape?: ButtonShape
+  icon?: IconFont
+  disabled?: boolean
+  status?: RemoteDataState
+  titleText?: string
+}
+
+class Button extends Component<Props> {
+  public static defaultProps: Partial<Props> = {
+    color: ComponentColor.Default,
+    size: ComponentSize.Small,
+    shape: ButtonShape.Default,
+    disabled: false,
+    status: RemoteDataState.Done,
+  }
+
+  public render() {
+    const {disabled, onClick, text, titleText} = this.props
+
+    return (
+      <button
+        className={this.className}
+        disabled={disabled}
```

Comment:
could `disabled` be something that is determined from `status`?

Reply:
I suppose you could include `disabled` as a type of `status` and that wouldn't be confusing